### PR TITLE
feat: Add DesignPoint plotting and button

### LINKED ISSUE #723: PLOTTING: enable representation of DesignPoints
DesignPoints would also be interesting to represent in the plotter. We should also have an option to either enable/disable the view of DesignPoints on the plotter. Just like our bounding box ruler widget.

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
Allows to plot DesignPoints, and to show/hide them in the plotter.
 - [x] Document new methods properly.
 - [x] Test methods that are testable.
 - [x] Add custom image for the button.

Closes #723

### CODE REVIEW

File: src/ansys/geometry/core/designer/designpoint.py [hunk 1]
```diff
@@ -89,4 +90,8 @@ def __repr__(self) -> str:
         return "\n".join(lines)
 
     def _to_polydata(self) -> pv.PolyData:
-        return pv.Sphere(center=self.value.flat, radius=0.001)
+        """Get polydata from DesignPoint object."""
+        # get units to plot proportionally
+        ureg = UnitRegistry()
+        unit = 1 * self.value.unit
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/designpoint.py:96]: Nope, use our UNITS object. Do not redeclare a ``UnitRegistry``. This will break things

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: src/ansys/geometry/core/designer/designpoint.py [hunk 2]
```diff
@@ -22,8 +22,10 @@
 """Module for creating and managing design points."""
 
 from beartype.typing import TYPE_CHECKING, Union
+import pyvista as pv
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/designpoint.py:25]: Perform the import inside the _to_polydata method if possible @AlejandroFernandezLuces - that will avoid loading PyVista unless strictly necessary for plotting

File: src/ansys/geometry/core/designer/designpoint.py [hunk 3]
```diff
@@ -22,8 +22,10 @@
...
 
 from ansys.geometry.core.math.point import Point3D
+from ansys.geometry.core.misc import UNITS
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/designpoint.py:28]: Search for the actual ``UNITS`` import, not the easy access. This is critical for the documentation to render properly.

File: src/ansys/geometry/core/designer/designpoint.py [hunk 4]
```diff
@@ -86,3 +88,9 @@ def __repr__(self) -> str:
         lines.append(f"  Name                 : {self.name}")
         lines.append(f"  Design Point         : {self.value}")
         return "\n".join(lines)
+
+    def _to_polydata(self) -> pv.PolyData:
+        """Get polydata from DesignPoint object."""
+        # get units to plot proportionally
+        unit = 0.3 * self.value.unit
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/designpoint.py:95]: 0.3 seems kind of a magic number. Can we explain the rationale?

File: src/ansys/geometry/core/plotting/plotter_helper.py [hunk 5]
```diff
@@ -224,8 +228,10 @@ def compute_edge_object_map(self) -> Dict[pv.Actor, EdgePlot]:
             Mapping between plotter actors and EdgePlot objects.
         """
         for object in self._geom_object_actors_map.values():
-            for edge in object.edges:
-                self._edge_actors_map[edge.actor] = edge
+            # get edges only from bodies
+            if isinstance(object, Body) or isinstance(object, MasterBody):
```
Comment (RobPasMue) [inline src/ansys/geometry/core/plotting/plotter_helper.py:232]: Faces would also be accepted. Add it now that you can :)

File: src/ansys/geometry/core/plotting/widgets/_images/designpoint.png
Comment (RobPasMue) [inline src/ansys/geometry/core/plotting/widgets/_images/designpoint.png:1]: Nice icon

File: src/ansys/geometry/core/plotting/widgets/measure.py [hunk 6]
```diff
@@ -30,12 +30,12 @@
...
 class MeasureWidget(PlotterWidget):
     """
-    Provides the ruler widget for the PyAnsys Geometry ``Plotter`` class.
+    Provides the measure widget for the PyAnsys Geometry ``Plotter`` class.
 
     Parameters
     ----------
-    plotter : ~pyvista.Plotter
-        Provides the plotter to add the ruler widget to.
+    plotter : PlotterHelper
```
Comment (RobPasMue) [inline src/ansys/geometry/core/plotting/widgets/measure.py:37]:
```suggestion
    plotter_helper : PlotterHelper
```

File: src/ansys/geometry/core/plotting/widgets/show_design_point.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+from ansys.geometry.core.plotting.widgets.widget import PlotterWidget
+
+
+class ShowDesignPoints(PlotterWidget):
+    """
+    Provides the a button to hide/show DesignPoint objects in the plotter.
+
+    Parameters
+    ----------
+    plotter : PlotterHelper
```
Comment (RobPasMue) [inline src/ansys/geometry/core/plotting/widgets/show_design_point.py:38]:
```suggestion
    plotter_helper : PlotterHelper
```

File: src/ansys/geometry/core/plotting/plotter_helper.py [hunk 8]
```diff
@@ -77,6 +79,7 @@ def __init__(
         self._picked_list = set()
         self._picker_added_actors_map = {}
         self._edge_actors_map = {}
+        self._design_point_actors = []
```
Comment (RobPasMue) [inline src/ansys/geometry/core/plotting/plotter_helper.py:82]: I don't see it used (I think)

**Review by RobPasMue — CHANGES_REQUESTED** [review]
Looking good @AlejandroFernandezLuces!

File: src/ansys/geometry/core/plotting/widgets/_images/designpoint.png
Comment (AlejandroFernandezLuces) [reply src/ansys/geometry/core/plotting/widgets/_images/designpoint.png:1]: FYI, I'm getting the icons from Material, they are open source: https://fonts.google.com/icons

File: src/ansys/geometry/core/plotting/plotter_helper.py (hunk 8 above)
Comment (AlejandroFernandezLuces) [reply src/ansys/geometry/core/plotting/plotter_helper.py]: It is unused, removing it

**Comment by AlejandroFernandezLuces** [comment]
Forgot to upload a gif of the plotter with the points, here it is:
Recording 2023-09-27 at 12 47 30

File: src/ansys/geometry/core/designer/designpoint.py (hunk 4 above)
Comment (AlejandroFernandezLuces) [reply src/ansys/geometry/core/designer/designpoint.py:96]: Honestly, I had to select a size for the sphere representation of the points. I think the default value of the sphere was too big, so I reduced its size. 

Anyway, we should be representing the points with a decimal of the original unit, so we don't hide full units under the sphere while keeping the representation of the point visible in the plotter.

I can't think of a better approach to this, if you have an idea of how to approach this better please let me know 🙂

**Comment by RobPasMue** [comment]
I have the feeling all these "viz" widgets have to be reordered. Measurement widget, design points widget, grid widget should all be placed together. Maybe on the right side of the plotter and we leave the view widgets on the left? What do you think?

**Comment by AlejandroFernandezLuces** [comment]
> I have the feeling all these "viz" widgets have to be reordered. Measurement widget, design points widget, grid widget should all be placed together. Maybe on the right side of the plotter and we leave the view widgets on the left? What do you think?

I agree. I would even would further and implement hide/show button to hide the widgets, or each group of widgets, and don't have them on display continuously. But I would leave it for another PR honestly.

**Comment by RobPasMue** [comment]
> I agree. I would even would further and implement hide/show button to hide the widgets, or each group of widgets, and don't have them on display continuously. But I would leave it for another PR honestly.

Agreed yeah - let's do a follow up PR for that. Thanks @AlejandroFernandezLuces

File: src/ansys/geometry/core/designer/designpoint.py (hunk 4 above)
Comment (RobPasMue) [reply src/ansys/geometry/core/designer/designpoint.py:96]: Adding a comment saying precisely this, is more than enough =)

> 0.3 = size for the sphere representation of the points - determined empirically for proper representation

File: tests/integration/test_plotter.py [hunk 9]
```diff
@@ -657,3 +657,27 @@ def test_name_filter(modeler: Modeler, verify_image_cache):
         filter="Cyl",
         screenshot=Path(IMAGE_RESULTS_DIR, "test_name_filter.png"),
     )
+
+
+def test_plot_design_point(modeler: Modeler, verify_image_cache):
```
Comment (RobPasMue) [inline tests/integration/test_plotter.py:662]: Ideally - add a docstring for the test. Also add it to the one for name filters. Just realized it was missing

**Review by RobPasMue — CHANGES_REQUESTED** [review]

**Pull request merged**